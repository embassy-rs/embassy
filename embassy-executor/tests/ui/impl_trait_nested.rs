#![cfg_attr(feature = "nightly", feature(impl_trait_in_assoc_type))]

struct Foo<T>(T);

#[embassy_executor::task]
async fn foo(_x: Foo<impl Sized + 'static>) {}

fn main() {}
