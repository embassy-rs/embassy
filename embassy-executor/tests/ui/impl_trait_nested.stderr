error: `impl Trait` is not allowed in task arguments. It is syntax sugar for generics, and tasks can't be generic.
 --> tests/ui/impl_trait_nested.rs:6:22
  |
6 | async fn foo(_x: Foo<impl Sized + 'static>) {}
  |                      ^^^^^^^^^^^^^^^^^^^^
